File: catalog/backends/java-quarkus/patterns/FeatureFlagBooleanGate.java
package com.example.service;

import com.datadoghq.openfeature.Provider;
import dev.openfeature.sdk.*;
import jakarta.annotation.PostConstruct;
import jakarta.enterprise.context.ApplicationScoped;

// Pattern: Feature Flags — boolean gate with user targeting context
// Deps: add to pom.xml:
//   com.datadoghq:dd-openfeature:1.57.0
//   dev.openfeature:sdk:1.18.2
// Adapt: replace "checkout-new-flow" with your flag key; add domain attributes to context
@ApplicationScoped
public class FeatureFlagBooleanGate {

    private Client client;

    @PostConstruct
    void init() {
        OpenFeatureAPI api = OpenFeatureAPI.getInstance();
        api.setProviderAndWait(new Provider());
        this.client = api.getClient();
    }

    public boolean isCheckoutEnabled(String userId, String plan) {
        EvaluationContext ctx = new MutableContext(userId).add("plan", plan);
        return client.getBooleanValue("checkout-new-flow", false, ctx);
    }
}

File: catalog/backends/python-django/patterns/feature_flag_boolean_gate.py
from ddtrace import tracer
from openfeature import api
from ddtrace.openfeature import DataDogProvider
from openfeature.evaluation_context import EvaluationContext

# Pattern: Feature Flags — boolean gate with user targeting context
# Deps: ddtrace>=3.19.0, openfeature-sdk>=0.5.0
# Adapt: replace "checkout-new-flow" with your flag key; add domain attributes to context

tracer.configure()
api.set_provider(DataDogProvider())
_client = api.get_client()


def is_checkout_enabled(user_id: str, plan: str) -> bool:
    ctx = EvaluationContext(targeting_key=user_id, attributes={"plan": plan})
    return _client.get_boolean_value("checkout-new-flow", False, ctx)
